File: testApp/core/source/org/openfaces/testapp/jBossSeam/messages/Message.java
/*
 * OpenFaces - JSF Component Library 2.0
 * Copyright (C) 2007-2012, TeamDev Ltd.
 * licensing@openfaces.org
 * Unless agreed in writing the contents of this file are subject to
 * the GNU Lesser General Public License Version 2.1 (the "LGPL" License).
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * Please visit http://openfaces.org/licensing/ for more details.
 */

//$Id: Message.java,v 1.3 2006/01/13 14:19:20 theute Exp $
package org.openfaces.testapp.jBossSeam.messages;

import org.hibernate.validator.Length;
import org.hibernate.validator.NotNull;
import static org.jboss.seam.ScopeType.EVENT;
import org.jboss.seam.annotations.Name;
import org.jboss.seam.annotations.Scope;

import javax.persistence.Basic;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import java.io.Serializable;
import java.util.Date;

@Entity
@Name("seammessage")
@Scope(EVENT)
public class Message implements Serializable {
    private Long id;
    private String title;
    private String text;
    private boolean read;
    private Date datetime;

    @Id
    @GeneratedValue
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @NotNull
    @Length(max = 100)
    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @NotNull
    @Lob
    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    @NotNull
    public boolean isRead() {
        return read;
    }

    public void setRead(boolean read) {
        this.read = read;
    }

    @NotNull
    @Basic
    @Temporal(TemporalType.TIMESTAMP)
    public Date getDatetime() {
        return datetime;
    }

    public void setDatetime(Date datetime) {
        this.datetime = datetime;
    }

}

File: openFaces/source/META-INF/resources/openfaces/validation/message.js
/*
 * OpenFaces - JSF Component Library 3.0
 * Copyright (C) 2007-2012, TeamDev Ltd.
 * licensing@openfaces.org
 * Unless agreed in writing the contents of this file are subject to
 * the GNU Lesser General Public License Version 2.1 (the "LGPL" License).
 * This library is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * Please visit http://openfaces.org/licensing/ for more details.
 */

O$._MessageRenderer = function(clientId, forClientId, title, tooltip, showSummary, showDetail, style, css, styles, params) {
  this.clientId = clientId;
  this.forId = forClientId;
  this.tooltip = tooltip;
  this.title = title;
  this.showSummary = showSummary;
  this.showDetail = showDetail;
  this.style = style;
  this.css = css;
  this.styles = styles;
  this.params = params;
};

O$.extend(O$._MessageRenderer.prototype, {
  update: function() {
    var spanElement = O$(this.clientId);
    if (!spanElement)
      return;
    while (spanElement.firstChild) {
      spanElement.removeChild(spanElement.firstChild);
      spanElement.title = null;
    }
    var forElement = O$.byIdOrName(this.forId);
    if (forElement) {
      var message = O$.getMessages(forElement)[0];
      if (message) {
        var summary = message.summary;
        var detail = message.detail;

        var severityStyle = this.getSeverityStyle(message.severity);
        var severityClass = this.getSeverityClass(message.severity);

        var title = this.title;
        var showSummaryAsTooltip = !title && this.tooltip;
        var showSummary = this.showSummary && summary;
        var showDetail = this.showDetail && detail;
        var summaryRenderedAsTitle = showSummaryAsTooltip && showSummary && showDetail;
        if (summaryRenderedAsTitle) {
          title = summary;
        }
        if (title) {
          spanElement.title = title;
        }
        var text = "";
        if (showSummary && !summaryRenderedAsTitle) {
          text = summary;
          if (showDetail) {
            text = text + " ";
          }
        }
        if (showDetail) {
          text = text + detail;
        }

        var textNode = document.createTextNode(text);

        spanElement.appendChild(textNode);
        if (spanElement.style.display == "none") {
          spanElement.style.display = "block";
        }
        if (severityStyle && severityStyle.length > 0) {
          spanElement.setAttribute("style", severityStyle);
        }
        if (severityClass && severityClass.length > 0) {
          spanElement.className = severityClass;
        }
        if (this.params) {
          for (var attr in this.params) {
            spanElement.setAttribute(attr, this.params[attr]);
          }
        }
      } else {
        if (spanElement.style.display == "block") {
          spanElement.style.display = "none";
        }
      }
    }
  },

  getSeverityStyle: function(severity) {
    if (!severity)
      return this.style;
    if (!this.styles)
      return this.style;

    var s = severity + "Style";

    for (var attr in this.styles) {
      if (attr.toLowerCase() == s.toLowerCase())
        return this.styles[attr];
    }
    return this.style;

  },

  getSeverityClass: function(severity) {
    if (!severity)
      return this.css;
    if (!this.styles)
      return this.css;

    var s = severity + "Class";

    for (var attr in this.styles) {
      if (attr.toLowerCase() == s.toLowerCase())
        return this.styles[attr];
    }
    return this.css;

  }

});